MSCL/source/mscl/MicroStrain/MIP/NMEA/NmeaPacketCollector.h

```diff
@@ -0,0 +1,85 @@
+/*****************************************************************************************
+**          Copyright(c) 2015-2022 Parker Hannifin Corp. All rights reserved.           **
+**                                                                                      **
+**    MIT Licensed. See the included LICENSE.txt for a copy of the full MIT License.    **
+*****************************************************************************************/
+
+#pragma once
+
+#include <condition_variable>
+#include <boost/circular_buffer.hpp>
+
+#include "NmeaPacket.h"
+
+namespace mscl
+{
+    //Class: NmeaPacketCollector
+    //    Class that sets/matches expected responses, and stores all the NMEA packets
+    class NmeaPacketCollector
+    {
+    public:
+        //Constructor: NmeaPacketCollector
+        //    The default constructor for the NmeaPacketCollector
+        NmeaPacketCollector();
+
+        //Destructor: NmeaPacketCollector
+        //    The destructor for the NmeaPacketCollector
+        ~NmeaPacketCollector();
+
+        NmeaPacketCollector(const NmeaPacketCollector&) = delete;             //copy constructor disabled
+        NmeaPacketCollector& operator=(const NmeaPacketCollector&) = delete;  //assignment operator disabled
+
+    private:
+        //Typedef: circular_data_buffer
+        //    Typedef for a circular buffer of a <NmeaPacket>
+        typedef boost::circular_buffer_space_optimized<NmeaPacket> circular_data_buffer;
+
+        //Constant: MAX_DATA_BUFFER_SIZE = 1024 * 100
+        //    The maximum number of data packets that can be stored in the circular buffer
+        static const int MAX_DATA_BUFFER_SIZE = 1024 * 100;
```

Comment:
Defining and initializing a `static const` variables in a header file tends to be a problem on real compilers (not Windows), so I would make this `static constexpr` or move the initialization into the `.cpp` file